lib/logger/lib/logger.ex

```diff
@@ -573,7 +573,7 @@ defmodule Logger do
     if compare_levels(level, min_level) != :lt do
       macro_log(level, data, metadata, caller)
     else
-      :ok
+      handle_unused_variable_warnings(data)
```

Comment:
We don't actually put pressure on logger because we read the level from the ETS table. So most of the cost will be in calculating whatever needs to be calculated? I would keep this patch but continue NOT calling functions and document the behaviour of purging better and call it a day.